crates/sui-framework/packages/dwallet-system/sources/dwallet_transfer.move

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) dWallet Labs, Ltd.
+// SPDX-License-Identifier: BSD-3-Clause-Clear
+
+#[allow(unused_use)]
+module dwallet_system::dwallet_transfer {
+    use dwallet::object::{Self, ID, UID};
+    use dwallet::token::recipient;
+    use dwallet::transfer;
+    use dwallet::tx_context;
+    use dwallet::tx_context::TxContext;
+    use dwallet_system::dwallet_2pc_mpc_ecdsa_k1::{DWallet, output};
+
+    const DWALLET_TRANSFER_ERROR: u64 = 0x1;
+
+    struct DwalletTransfer has key{
+        id: UID,
+        dwallet_id: ID,
+        encrypted_secret_share: vector<u8>,
+    }
+
+    struct PublicKey has key {
+        id: UID,
+        public_key: vector<u8>,
```

Comment:
Please add a `scheme` member that is an enum, that you now only have one option `Paillier`. Make sure to send this as a parameter to the `verify_dwallet_transfer` function. This is so that in the future we can use other algorithms

Reply:
I'm not sure if it is necessary to send the `scheme` to the `verify_dwallet_transfer` function at this stage. Perhaps there is a better design we can consider, and make it generic in a separate task?

Reply:
Also, it seems like there are no emuns in move